core/storage/collection/gae_models_test.py

```diff
@@ -0,0 +1,208 @@
+        commits = collection_models.CollectionSummaryModel.get_non_private()
+        self.assertEqual(1, len(commits))
+
+    def test_get_private_at_least_viewable(self):
+        viewable_collection_summary_model = (
+            collection_models.CollectionSummaryModel(
+                id='id0',
+                title='title',
+                category='category',
+                objective='objective',
+                language_code='language_code',
+                tags=['tags'],
+                status=constants.ACTIVITY_STATUS_PRIVATE,
+                community_owned=False,
+                owner_ids=['owner_ids'],
+                editor_ids=['editor_ids'],
+                viewer_ids=['a'],
+                contributor_ids=[''],
+                contributors_summary={},
+                version=0,
+                node_count=0,
+                collection_model_last_updated=None,
+                collection_model_created_on=None,
+            ))
+        viewable_collection_summary_model.put()
+
+        unviewable_collection_summary_model = (
+            collection_models.CollectionSummaryModel(
+                id='id1',
+                title='title',
+                category='category',
+                objective='objective',
+                language_code='language_code',
+                tags=['tags'],
+                status=constants.ACTIVITY_STATUS_PRIVATE,
+                community_owned=False,
+                owner_ids=['owner_ids'],
+                editor_ids=['editor_ids'],
+                viewer_ids=['viewer_ids'],
+                contributor_ids=[''],
+                contributors_summary={},
+                version=0,
+                node_count=0,
+                collection_model_last_updated=None,
+                collection_model_created_on=None,
+            ))
+        unviewable_collection_summary_model.put()
+        commits = (
+            collection_models.CollectionSummaryModel
+            .get_private_at_least_viewable('a'))
+        self.assertEqual(1, len(commits))
+
+    def test_get_at_least_editable(self):
+        editable_collection_summary_model = (
+            collection_models.CollectionSummaryModel(
+                id='id0',
+                title='title',
+                category='category',
+                objective='objective',
+                language_code='language_code',
+                tags=['tags'],
+                status=constants.ACTIVITY_STATUS_PRIVATE,
+                community_owned=False,
+                owner_ids=['a'],
+                editor_ids=['editor_ids'],
+                viewer_ids=['viewer_ids'],
+                contributor_ids=[''],
+                contributors_summary={},
+                version=0,
+                node_count=0,
+                collection_model_last_updated=None,
+                collection_model_created_on=None,
+            ))
+        editable_collection_summary_model.put()
+
+        uneditable_collection_summary_model = (
```

Comment:
Maybe we can delete this one and test somewhat like this:
```
collections =  (collection_models.CollectionSummaryModel
           .get_at_least_editable('a'))
       self.assertEqual(1, len(collections))
collections =  (collection_models.CollectionSummaryModel
           .get_at_least_editable('viewers_ids'))
       self.assertEqual(0, len(collections))
```

Reply:
Done. This seems like a good way to test more functionality.